documentation/config_logging_advanced.md

```diff
@@ -4,123 +4,131 @@ title: "Puppet Server: Advanced Logging Configuration"
+    ``` xml
+    <appender name="JSON" class="ch.qos.logback.core.rolling.RollingFileAppender">
+            <file>/var/log/puppetlabs/puppetserver/puppetserver.log.json</file>
 
-        <encoder class="net.logstash.logback.encoder.LogstashEncoder"/>
-    </appender>
-```
+            <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+                <fileNamePattern>/var/log/puppetlabs/puppetserver/puppetserver.log.json.%d{yyyy-MM-dd}</fileNamePattern>
+                <maxHistory>5</maxHistory>
+            </rollingPolicy>
 
-Logback writes logs using components called [appenders](http://logback.qos.ch/manual/appenders.html).  Note that the above configuration includes the `RollingFileAppender`, to rotate the log files and ensure they don't fill up your disk.  Also, note that there are a lot of configuration options available for the `LogstashEncoder` above, including the ability to modify the list of fields that you wish to include, or give them different field names.  For more info see the [Logstash Logback Encoder Docs](https://github.com/logstash/logstash-logback-encoder/blob/master/README.md#loggingevent-fields).
+            <encoder class="net.logstash.logback.encoder.LogstashEncoder"/>
+        </appender>
+    ```
 
-The above XML snippet will create an appender that will log to the specified JSON file when active.  In order to activate the appender, you also need to add an `appender-ref`, e.g. in the `<root>` section of your `logback.xml`:
+2. Activate the appended by adding an `appender-ref` entry to the `<root>` section of `logback.xml`:
 
-``` xml
-<root level="info">
-        <appender-ref ref="FILE"/>
-        <appender-ref ref="JSON"/>
-</root>
-```
+    ``` xml
+    <root level="info">
+            <appender-ref ref="FILE"/>
+            <appender-ref ref="JSON"/>
+    </root>
+    ```
 
-You could also comment out or remove the other `appender-ref`s, if you decided you *only* wanted to log the JSON format.
+3. If you decide you want to log *only* the JSON format, comment out the other `appender-ref` entries.
+
+`LogstashEncoder` has many configuration options, including the ability to modify the list of fields that you want to include, or give them different field names. For more information, see the [Logstash Logback Encoder Docs](https://github.com/logstash/logstash-logback-encoder/blob/master/README.md#loggingevent-fields).
 
 #### Adding a JSON version of the Puppet Server HTTP Access logs
 
-To add JSON logging for Puppet Server's HTTP requests, modify the `request-logging.xml` file.  Here's an example of adding a logback appender there:
+To add JSON logging for HTTP requests:
+
+1. Add the following Logback appender section to the `request-logging.xml` file:
 
-``` xml
-{% raw %}
-   <appender name="JSON" class="ch.qos.logback.core.rolling.RollingFileAppender">
-        <file>/var/log/puppetlabs/puppetserver/puppetserver-access.log.json</file>
+    ``` xml
+    {% raw %}
```

Comment:
The raw tags are Liquid tags for the Jekyll static site builder. We've had problems with Jekyll interpreting parts of the code that are similar to Liquid tags; ``{% raw %}`` tells Jekyll to ignore any Liquid tags between them. This ensures the code example renders correctly on the docs site.